dstep/driver/Application.d

```diff
@@ -26,280 +26,164 @@ import clang.Index;
 import clang.TranslationUnit;
 import clang.Util;
 
+import dstep.Configuration;
+import dstep.translator.Options;
 import dstep.core.Exceptions;
 import dstep.translator.Translator;
 
-class Application : DStack.Application
+class Application
 {
-    mixin Singleton;
-
     enum Version = "0.2.2";
 
     private
     {
-        Language language;
-        string[] argsToRestore;
-        bool helpFlag;
-        Compiler compiler;
+        AppConfig config;
     }
 
-    protected override void run ()
+    public this ( AppConfig config )
     {
-        handleArguments;
-
-        if (arguments.argument.input.hasValue)
-        {
-            auto inputFiles = arguments.argument.input.values[1 .. $];
-            string outputDir = null;
-            if (inputFiles.length > 0)
-            {
-                outputDir = arguments.output;
-                if (arguments.output.any() && !exists(outputDir))
-                    mkdirRecurse(outputDir);
-            }
-
-            foreach(string fileName; inputFiles)
-            {
-                auto conversionTask = task!startParsingFile(language, argsToRestore, fileName, true,
-                                                            arguments, compiler, compilerArgs, outputDir);
-                conversionTask.executeInNewThread();
-            }
-
-            startParsingFile(language, argsToRestore, arguments.argument.input.values[0], false, arguments,
-                             compiler, compilerArgs, outputDir);
-        }
-
-        else
-            startParsingFile(language, argsToRestore, "", false, arguments, compiler, compilerArgs, null);
+        this.config = config;
     }
 
-    protected override void setupArguments ()
+    public void run ()
     {
-        arguments.sloppy = true;
-        arguments.passThrough = true;
+        import std.exception : enforce;
 
-        arguments.argument("input", "input files").params(1, int.max);
+        enforce!DStepException(config.input_files.length > 0,
+            "Must supply at least one input file");
 
-        arguments('o', "output", "Write output to")
-            .params(1)
-            .defaults("");
+        // when only one input file is supplied, -o argument is
+        // interpreted as file path, otherwise as base directory path
+        bool singleFileInput = config.input_files.length == 1;
 
-        arguments('x', "language", "Treat subsequent input files as having type <language>")
-            .params(1)
-            .restrict("c", "c-header", "objective-c", "objective-c-header")
-            .on(&handleLanguage);
-
-        arguments("no-comments", "Disable translation of comments.");
+        // parallel generation of D modules for each of input files
+        foreach(string fileName; config.input_files)
```

Comment:
* Please but a space between `foreach` and the opening parenthesis
* Use type inference, i.e. remove `string`